arches_lingo/src/arches_lingo/components/generic/reference-datatype/ReferenceDatatypeEditor.vue

```diff
@@ -0,0 +1,97 @@
+<script setup lang="ts">
+import { computed, inject, toRef } from "vue";
+import Select from "primevue/select";
+import MultiSelect from "primevue/multiselect";
+
+import { systemLanguageKey } from "@/arches_lingo/constants.ts";
+import { PREF_LABEL } from "@/arches_vue_utils/constants.ts";
+
+import type { ControlledListItem } from "@/arches_lingo/types";
+import type { Language } from "@/arches_vue_utils/types.ts";
+
+const systemLanguage = inject(systemLanguageKey) as Language;
+const emit = defineEmits(["update"]);
+const props = withDefaults(
+    defineProps<{
+        value?: ControlledListItem[];
+        options?: ControlledListItem[];
+        multiValue?: boolean;
+    }>(),
+    {
+        value: () => [],
+        options: () => [],
+    },
+);
+
+const modelValue = toRef(props, "value");
+const val = computed({
+    get() {
+        return extractURI(modelValue.value);
+    },
+    set(newVal) {
+        if (!props.multiValue) {
+            const foundItem = props.options?.find(
+                (item) => item.uri === newVal,
+            );
+            const item = foundItem ? [foundItem] : [];
+            emit("update", item);
+        } else {
+            const foundItems = props.options?.filter((item) =>
+                newVal.includes(item.uri),
+            );
+            const items = foundItems ?? [];
+            emit("update", items);
+        }
+    },
+});
+
+function extractURI(item: ControlledListItem[]): string | string[] {
+    if (item && !props.multiValue) {
+        return item[0]?.uri;
+    } else if (item && props.multiValue) {
+        return item.map((item) => item.uri);
+    } else {
+        return [];
+    }
+}
+
+function getOptionLabels(item: ControlledListItem): string {
+    const prefLabels = item.labels.filter(
+        (label) => label.valuetype_id === PREF_LABEL,
+    );
+    const optionLabel =
+        prefLabels.find(
+            (label) => label.language_id === systemLanguage?.code,
+        ) || prefLabels[0];
+    return optionLabel?.value ?? "";
+}
+</script>
+
+<template>
+    <Select
+        v-model="val"
+        v-if:="!props.multiValue"
+        :show-toggle-all="options?.length"
```

Comment:
Although I use `.length` in boolean contexts (maybe more than I should), this is a prop doc'd to take a boolean so let's actually pass a boolean, even if it's not happening to break right now.
```suggestion
        :show-toggle-all="options?.length > 0"
```

Reply:
<img width="259" alt="Screenshot 2025-01-08 at 2 09 18 PM" src="https://github.com/user-attachments/assets/6eaffb7a-4855-4af9-b254-f080788237f1" />

Reply:
would say if we're just casting to a bool then `Boolean()` is more explicit